### ISSUE TITLE
Support an image along with the title in search results

### ISSUE THREAD

**Issue by martindholmes** [projectEndings/staticSearch#49]
For a couple of projects, including MyNDIR, it'll be important to be able to do a text and filter search for metadata documents that describe images; and for these cases, which are not unusual, I think it would make sense to be able to provide the path to an image which would be presented alongside the document title in a search hit.

This could be configured by adding a meta tag like the one that overrides the default document title, something like this:

~~~~~
<meta name="docImage" class="staticSearch.docImage" content="images/thisPage.jpg" />
~~~~~

This information could be harvested into the ssTitles*.json file, so that the search engine could provide the image (linked to the document) alongside the title in the result set.

Things to decide: 

1. If the image path is relative, what is it relative to? Presumably the search page?
1. Should the build process check that linked images are there?

**Comment by joeytakeda** [comment]
Excellent idea. This will be useful for WEA, too, which has a facsimile thumbnail for every primary source document.

> If the image path is relative, what is it relative to? Presumably the search page?

I think relative to the search page is okay, since we're using the `@content` for the image path, which, in that context, is just a helpful string hint; so long as we're not bound by some rule in the HTML spec that says "If the content looks like a path thing, then it should be resolved as a path", then we're probably fine to resolve the `@content` relative to the search page.

> Should the build process check that linked images are there?

Yes that's probably worth it. We should probably add it to the reports page as a diagnostic with a little bit of preceding text saying something to the effect of "By the way, the image path is relative to the SEARCH page not the document". I suppose the process would be something like:

* Gather the entire set of paths (i.e. `images/blort.jpg`, `otherImageDir/foo.png`, `someOtherDirEntirely/image.png`)
* Get the unique values of all of the directory paths
* Create a sequence of all of the URI collections of all of the directory paths (`$imageUris`)
* Check to see if each image path is resolvable to something in `$imageUris`

**Comment by martindholmes** [comment]
Do we insist that the supplied image path is relative to the search page (which would make it potentially incorrect relative to its host document), or do we insist that it's relative to the host document, and we then resolve it relative to the search page during the build process?

**Comment by joeytakeda** [comment]
I think the `<meta>` being relative to the host document and then resolved to be relative to the search page in processing is probably better practice, even though it is a bit trickier (though I suppose it's not all that tricky since we capture the doc's uri right away in the process).

**Comment by martindholmes** [comment]
@joeytakeda This is at the top of my list at the moment, because of the MyNDIR rewrite, so I've started work on it in a separate branch. So far I've added documentation, and code to add the thumbnail as a string value in the ssTitles*.json file, but this is done on the assumption that the path has been rewritten to be relative to the search page during the tokenizing phase. If you think this is the wrong approach, shout and I'll rethink it.
